infrastructure/smart-contracts/contracts/mock/MockRollup.sol

```diff
@@ -0,0 +1,143 @@
+        bytes32 wasmModuleRoot = keccak256(abi.encodePacked(blockhash(block.number - 5), block.timestamp, msg.sender));
+        
+        // Compute the simulated confirmData hash using the provided blockHash and sendRoot
+        bytes32 confirmData = keccak256(abi.encodePacked(blockHash, sendRoot));
+        
+        // Initialize an assertion structure with dummy data this data will not be used in this mock contract
+        Assertion memory assertion = Assertion(
+            GlobalState({bytes32Vals: bytes32(0), u64Vals: [uint64(0), uint64(0)]}),
+            GlobalState({bytes32Vals: bytes32(0), u64Vals: [uint64(0), uint64(0)]}),
+            uint64(0)
+        );
+
+        // Store the newly created node in the mapping with all its attributes
+        _nodes[latestNodeCreated] = Node({
+            stateHash: bytes32(0),             // Placeholder: In practice, this should represent the state of the node
+            challengeHash: bytes32(0),         // Placeholder: Represents data that can be challenged
+            confirmData: confirmData,          // Store the computed confirmData hash for validation during confirmation
+            prevNum: latestNodeCreated,        // Set previous node index to the latest node created before this one
+            deadlineBlock: 0,                  // Placeholder: Not used here
+            noChildConfirmedBeforeBlock: 0,    // Placeholder: Not used here
+            stakerCount: 0,                    // Placeholder: Not used here
+            childStakerCount: 0,               // Placeholder: Not used here
+            firstChildBlock: 0,                // Placeholder: Not used here
+            latestChildNumber: 0,              // Placeholder: Not used here
+            createdAtBlock: uint64(block.number),      // Record the block number at creation time for reference
+            nodeHash: nodeHash                 // Placeholder: Not used here
+        });
+
+        latestNodeCreated = nodeNum;
+
+        // Emit an event to notify that a new node has been created with all relevant details
+        emit NodeCreated(
+            latestNodeCreated,
+            parentNodeHash,
+            nodeHash,
+            executionHash,
+            assertion,
+            afterInboxBatchAcc,
+            wasmModuleRoot,
+            0
+        );
+    }
+
+    /**
+     * @notice Confirms a node by its number, validating against provided block hash and send root.
+     * @param nodeNum The number of the node to confirm.
+     * @param blockHash The simulated block hash to validate against the node's confirmData.
+     * @param sendRoot The simulated send root to validate against the node's confirmData.
+     */
+    function confirmNode(uint64 nodeNum, bytes32 blockHash, bytes32 sendRoot) external {
+        // Check that the node number is within the valid range of created nodes
+        require(nodeNum <= latestNodeCreated, "Invalid node number");
```

Comment:
Why is this require needed here ?
Seems like we are jsut making it ahrder for us to create nodes in a different order ?

Reply:
I assumed they should be confirmed in order, but I agree this makes no difference here. This has been removed and re-deployed!